connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/model/request/ProviderConfiguration.java

```diff
@@ -0,0 +1,198 @@
+ */
+package io.camunda.connector.agenticai.aiagent.model.request;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.camunda.connector.aws.model.impl.AwsAuthentication;
+import io.camunda.connector.feel.annotation.FEEL;
+import io.camunda.connector.generator.dsl.Property;
+import io.camunda.connector.generator.java.annotation.TemplateDiscriminatorProperty;
+import io.camunda.connector.generator.java.annotation.TemplateProperty;
+import io.camunda.connector.generator.java.annotation.TemplateSubType;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes({
+  @JsonSubTypes.Type(
+      value = ProviderConfiguration.AnthropicProviderConfiguration.class,
+      name = "anthropic"),
+  @JsonSubTypes.Type(
+      value = ProviderConfiguration.BedrockProviderConfiguration.class,
+      name = "bedrock"),
+  @JsonSubTypes.Type(
+      value = ProviderConfiguration.OpenAiProviderConfiguration.class,
+      name = "openai")
+})
+@TemplateDiscriminatorProperty(
+    label = "Provider",
+    group = "model",
+    name = "type",
+    description = "Specify the model provider to use")
+public sealed interface ProviderConfiguration
+    permits ProviderConfiguration.AnthropicProviderConfiguration,
+        ProviderConfiguration.BedrockProviderConfiguration,
+        ProviderConfiguration.OpenAiProviderConfiguration {
+
+  @TemplateSubType(id = "anthropic", label = "Anthropic")
+  record AnthropicProviderConfiguration(AnthropicConnection anthropic)
+      implements ProviderConfiguration {
+
+    public record AnthropicConnection(
+        @TemplateProperty(
+                group = "model",
+                description = "Specify endpoint if need to use a custom API endpoint",
+                type = TemplateProperty.PropertyType.Hidden,
+                feel = Property.FeelMode.disabled,
+                optional = true)
+            String endpoint,
+        AnthropicAuthentication authentication,
+        AnthropicModel model) {}
+
+    public record AnthropicAuthentication(
+        @NotBlank
+            @TemplateProperty(
+                group = "authentication",
+                label = "Anthropic API Key",
+                type = TemplateProperty.PropertyType.String,
+                feel = Property.FeelMode.optional)
+            String apiKey) {}
+
+    public record AnthropicModel(
+        @NotBlank
+            @TemplateProperty(
+                group = "model",
+                label = "Model",
+                type = TemplateProperty.PropertyType.String,
+                feel = Property.FeelMode.optional,
+                defaultValue = "claude-3-5-sonnet-20240620",
+                defaultValueType = TemplateProperty.DefaultValueType.String)
+            String model,
+        @Valid ModelParameters parameters) {}
```

Comment:
We are using the same parameters record for all 3 configurations, but it might be needed to have a specific implementation, based on the model, so I kept this under the specific model structure.